RPi4.md

```diff
@@ -0,0 +1,224 @@
+
+If all went well, you should see the commands return something like this:
+
+![RPI4 Console](/img/RPI4Terminal2.png)
+
+Next,  enable SSH at bootup so you can remotely login, and finally eject the  SD card so you can move it to the Raspberry Pi. The new SD card’s boot  partition should automatically be mounted on `/Volumes/boot`
+
+```
+touch /Volumes/boot/ssh
+diskutil eject disk2
+```
+
+![RPI4 Console](/img/RPI4Terminal4.png)
+
+## Booting up the Raspberry Pi
+
+After inserting the SD card into the Raspberry Pi, go ahead and connect the power and ethernet, and optionally the display and keyboard if you have those. It should boot up and get an IP address using DHCP. You can try searching for it with `ping raspberrypi.local` on your desktop PC, but if that doesn’t work you will need to login to your router to find its IP address.
+
+The IP address that my Raspberry Pi got was 192.168.1.5 so I SSH’d to that
+
+```
+ssh 192.168.1.5 -l pi
+```
+
+The  default password for the “pi” user is “raspberry”. After SSH’ing in, the first thing I want to do is check the device’s CPU temperature to  make sure the cooling system are working correctly:
+
+```
+sudo -svcgencmd measure_temp
+```
+
+Next, let’s change the password for the “pi” user.
+
+```
+passwd pi
+```
+
+![](C:\Users\Pavle\Documents\GitHub\btcpayserver-doc\btcpayserver-doc\btcpayserver-doc\img\RPI4Terminal4.png)
+
+I also recommend to disable swap to prevent burning out your SD card.
+
+```
+dphys-swapfile swapoff
+dphys-swapfile uninstall
+update-rc.d dphys-swapfile remove
+systemctl disable dphys-swapfile
+```
+
+![RPI4 Console](/img/RPI4Terminal5.png)
+
+Partition your SSD
+
+```
+fdisk /dev/sdatype 'p' to list existing partitions
+type 'd' to delete currently selected partitions
+type 'n' to create a new partition
+type 'w' to write the new partition table and exit fdisk
+```
+
+Format the new partition on your SSD
+
+```
+mkfs.ext4 /dev/sda1
+```
+
+Configure the SSD partition to auto-mount at bootup
+
+```
+mkfs.ext4 /dev/sda1
+mkdir /mnt/usb
+UUID="$(sudo blkid -s UUID -o value /dev/sda1)"
+echo "UUID=$UUID /mnt/usb ext4 defaults,noatime,nofail 0" | sudo tee -a /etc/fstab
+mount -a
+```
+
+While you’re editing `/etc/fstab` add a RAM filesystem for logs (optional). This is also to prevent burning out your SD card too quickly.
+
+```
+none        /var/log        tmpfs   size=10M,noatime         00
+```
+
+Mount the SSD partition and create a symlink for docker to use the SSD
+
+```
+mkdir /mnt/usb/docker
+ln -s /mnt/usb/docker /var/lib/docker
+```
+
+Upgrade your OS packages to latest
+
+```
+apt update && apt upgrade -y && apt autoremove
+```
+
+Install a firewall and allow SSH, HTTP, HTTPS, Bitcoin, and Lightning
+
+```
+apt install -y ufw
+ufw default deny incoming
+ufw default allow outgoing
+ufw allow 22/tcp
```

Comment:
This allows SSH connections from anywhere, which should not be the recommended option. Even though it's more error-prone (see my comment in the RPI3 section on this), we should restrict this to the local network. (i.e. `ufw allow from `192.168.1.0/24 to any port 22`)

Reply:
Ping @wiz thoughts?

Reply:
Firewall rules are probably out of scope for BTCPay installation anyway